drivers/i2c/i2c_mcux_lpi2c.c

```diff
@@ -161,6 +163,12 @@ static int mcux_lpi2c_transfer(struct device *dev, struct i2c_msg *msgs,
 			return -EIO;
 		}
 
+		k_busy_wait(DELAY_TIME(config->bitrate));
```

Comment:
If you do this wait only when `msgs->len == 0` does that fix the problem?  Because that's the case that isn't being handled correctly by the HAL.

I'd still like to see the delay duration based on something that makes sense, say a specific number of cycles of the I2C clock.

Reply:
Hi， Yes ， I add one if(msgs->len == 0), it works. Now the delay time is beyond CPU_FREQ & i2c's baudrate.